bin/apply-promotion-batch.py

```
@@ -0,0 +1,180 @@
+        print("DRY RUN — store untouched")
+        return 0
+
+    stamp = datetime.now(timezone.utc).strftime("%Y%m%dT%H%M%SZ")
+    backup = facts_path.with_name(f"facts.json.bak-preapply-{stamp}")
+    backup.write_bytes(facts_path.read_bytes())
+    facts_path.write_text(json.dumps(store, ensure_ascii=False), encoding="utf-8")
+
+    # Re-sign and strict-verify with the store's own toolchain; the batch is
+    # recorded applied ONLY after verification passes.
+    for script in ("sign-facts.py", "verify-facts.py"):
+        proc = subprocess.run(  # nosec B603 - fixed sibling scripts
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)